File: Modules/ServerTypes/hospital/DocSpecColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { DocSpecRow } from "./DocSpecRow";

export interface DocSpecColumns {
    DocSpecId: Column<DocSpecRow>;
    SpecialityId: Column<DocSpecRow>;
    DoctorId: Column<DocSpecRow>;
}

export class DocSpecColumns extends ColumnsBase<DocSpecRow> {
    static readonly columnsKey = 'hospital.DocSpec';
    static readonly Fields = fieldsProxy<DocSpecColumns>();
}

File: Modules/hospital/DocSpec/DocSpecColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace Serene2.hospital.Columns;

[ColumnsScript("hospital.DocSpec")]
[BasedOnRow(typeof(DocSpecRow), CheckNames = true)]
public class DocSpecColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int DocSpecId { get; set; }
    public int SpecialityId { get; set; }
    public int DoctorId { get; set; }
}